src/ert/_c_wrappers/enkf/analysis_config.py

```diff
@@ -60,17 +60,17 @@ def _copy_modules(self):
                     f"Trying to copy module {src_name!r} which does not exist"
                 )
 
-    def _set_modules_var_list(self):
+    def _set_modules_var_list(self) -> None:
         for set_var in self._analysis_set_var:
             module_name, var_name, value = set_var
 
             module = self.get_module(module_name)
             module.set_var(var_name, value)
 
     @classmethod
-    def from_dict(cls, config_dict) -> "AnalysisConfig":
-        num_realization = config_dict.get(ConfigKeys.NUM_REALIZATIONS, 1)
-        min_realization = config_dict.get(ConfigKeys.MIN_REALIZATIONS, 0)
+    def from_dict(cls, config_dict: ConfigDict) -> "AnalysisConfig":
+        num_realization: int = config_dict.get(ConfigKeys.NUM_REALIZATIONS, 1)  # type: ignore
```

Comment:
Some of these are Context'ed but some tests also initialize from dict without originating from file, so not sure which one is more practical/correct. Context* if file origin is always assumed.

![Screenshot 2023-07-27 at 15 48 03](https://github.com/equinor/ert/assets/11595637/71078970-3ad5-464b-8f86-e6618d711d7b)